File: Baekjoon/S5/C#/S5_그룹 단어 체커.cs
https://www.acmicpc.net/problem/1316

using System;

namespace Baekjoon
{
    class Program
    {
        static void Main(string[] args)
        {
            int n = int.Parse(Console.ReadLine());
            int ctr = 0;

            for (int i = 0; i < n; i++)
            {
                string str = Console.ReadLine();
                int[] charArr = new int[26];
                bool[] boolArr = new bool[26];
                bool check = true;

                for (int j = 0; j < str.Length; j++)
                {
                    if (boolArr[str[j] - 97] == false && charArr[str[j] - 97] > 0)
                    {
                        check = false;
                        break;
                    }
                    charArr[str[j] - 97]++;
                    boolArr[str[j] - 97] = true;
                    if (j != str.Length - 1 && boolArr[str[j] - 97] != boolArr[str[j + 1] - 97])
                        boolArr[str[j] - 97] = false;
                }

                if (check) ctr++;
            }

            Console.WriteLine(ctr);
        }
    }
}

File: Baekjoon/S5/JAVA/S5_그룹 단어 체커.java
//https://www.acmicpc.net/problem/1316

import java.io.IOException;
import java.io.BufferedReader;
import java.io.InputStreamReader;

public class Main{
	public static void main(String[] args) throws IOException{
	    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
	    int n = Integer.parseInt(br.readLine());
	    int cnt = n;
	    
	    while(n-- > 0){
	        String str = br.readLine();
			boolean[] arr = new boolean[26];
			
			for (int i = 0; i < str.length(); i++) {
				if(!arr[str.charAt(i) - 'a'])
					arr[str.charAt(i) - 'a'] = true;
					
				else if(i > 0 && str.charAt(i) != str.charAt(i-1)){
					cnt--;
					break;
				}
			}
	    }
	    
	    System.out.println(cnt);
	    br.close();
	}
}
